### PR TITLE
fix: cache issues (in event pages)

### PR DESCRIPTION
**Pull request by nikomakela**
HH-239. ApolloProviders were used wrongly

### CODE REVIEW

File: apps/hobbies-helsinki/src/domain/clients/eventsApolloClient.ts [hunk 1]
```diff
@@ -51,6 +64,14 @@ export function createEventsApolloClient() {
 
   const cache = createEventsApolloCache();
 
+  // if (isClient) {
```
Comment (nikomakela) [inline apps/hobbies-helsinki/src/domain/clients/eventsApolloClient.ts:67]: remove

File: apps/hobbies-helsinki/src/pages/courses/[eventId]/index.tsx [hunk 2]
```diff
@@ -50,7 +53,6 @@ export async function getStaticPaths() {
 }
 
 export async function getStaticProps(context: GetStaticPropsContext) {
-  const eventsApolloClient = createEventsApolloClient();
```
Comment (nikomakela) [inline apps/hobbies-helsinki/src/pages/courses/[eventId]/index.tsx:53]: check if there are more of these - should not be

File: packages/components/src/apollo/utils.ts [hunk 3]
```diff
@@ -87,7 +87,7 @@ export function initializeApolloClient<
   initialState,
   mutableCachedClient,
   createClient,
-}: InitApolloClientConfig<TCacheShape, Client>) {
+}: InitApolloClientConfig<TCacheShape, Client>): ApolloClient<TCacheShape> {
```
Comment (nikomakela) [inline packages/components/src/apollo/utils.ts:90]: should be needless?

File: apps/hobbies-helsinki/src/pages/_app.tsx [hunk 4]
```diff
@@ -57,20 +56,14 @@ export type AppProps<P = any> = {
...
   error: any;
 } & SSRConfig;
 
 function MyApp({ Component, pageProps }: AppProps<CustomPageProps>) {
-  const {
-    initialApolloState: initialCmsApolloState,
-    initialEventsApolloState,
-    error,
-  } = pageProps;
-  const cmsApolloClient = useCmsApollo(initialCmsApolloState);
```
Comment (nikomakela) [inline apps/hobbies-helsinki/src/pages/_app.tsx:72]: note: this was a really bad idea. It cleared the cache

File: apps/hobbies-helsinki/src/pages/_app.tsx [hunk 5]
```diff
@@ -57,20 +56,14 @@ export type AppProps<P = any> = {
...
-  const eventsApolloClient = useEventsApolloClient(initialEventsApolloState);
+  const { error } = pageProps;
+  const cmsApolloClient = useCmsApollo();
+  const eventsApolloClient = useEventsApolloClient();
```
Comment (nikomakela) [inline apps/hobbies-helsinki/src/pages/_app.tsx:66]: note: mutable reference of client has the right cache always

File: apps/hobbies-helsinki/src/pages/_app.tsx [hunk 6]
```diff
@@ -96,22 +89,20 @@ function MyApp({ Component, pageProps }: AppProps<CustomPageProps>) {
       <RHHCConfigProvider config={rhhcConfig}>
         <TopProgressBar />
         <ApolloProvider client={cmsApolloClient}>
-          <ApolloProvider client={eventsApolloClient}>
```
Comment (nikomakela) [inline apps/hobbies-helsinki/src/pages/_app.tsx:99]: I need to test this removal once more. At least it does not make sense to clear the context set by cmsClient!!

Comment (nikomakela) [reply apps/hobbies-helsinki/src/pages/_app.tsx:99]: I think it's better like this than to add a client to every query-hook. Less code and the component is more movable.

**Review by melniiv — APPROVED** [review]

**Pull request merged**